core/editor/src/tool/tools/select.rs

```diff
@@ -62,109 +66,110 @@ impl Default for SelectToolFsmState {
 		if let ToolMessage::Select(event) = event {
 			match (self, event) {
 				(Ready, DragStart) => {
 					data.drag_start = input.mouse.position;
 					data.drag_current = input.mouse.position;
-
-					let (point_1, point_2) = {
-						let (x, y) = (data.drag_start.x as f64, data.drag_start.y as f64);
-						(
-							DVec2::new(x - SELECTION_TOLERANCE, y - SELECTION_TOLERANCE),
-							DVec2::new(x + SELECTION_TOLERANCE, y + SELECTION_TOLERANCE),
-						)
-					};
-
-					let quad = [
-						DVec2::new(point_1.x, point_1.y),
-						DVec2::new(point_2.x, point_1.y),
-						DVec2::new(point_2.x, point_2.y),
-						DVec2::new(point_1.x, point_2.y),
-					];
-
-					if let Some(intersection) = document.document.intersects_quad_root(quad).last() {
-						// TODO: Replace root transformations with functions of the transform api
-						let transformed_start = document.document.root.transform.inverse().transform_vector2(data.drag_start.as_dvec2());
-						if document.layer_data.get(intersection).map_or(false, |layer_data| layer_data.selected) {
-							data.layers_dragging = document
-								.layer_data
-								.iter()
-								.filter_map(|(path, layer_data)| {
-									layer_data
-										.selected
-										.then(|| (path.clone(), document.document.layer(path).unwrap().transform.translation - transformed_start))
-								})
-								.collect();
-						} else {
-							responses.push_back(DocumentMessage::SelectLayers(vec![intersection.clone()]).into());
-							data.layers_dragging = vec![(intersection.clone(), document.document.layer(intersection).unwrap().transform.translation - transformed_start)]
+					let mut selected: Vec<_> = document.selected_layers().cloned().collect();
+					let quad = data.selection_quad();
+					let intersection = document.document.intersects_quad_root(quad);
+					// If no layer is currently selected and the user clicks on a shape, select that.
+					if selected.is_empty() {
+						if let Some(layer) = intersection.last() {
+							selected.push(layer.clone());
+							responses.push_back(DocumentMessage::SelectLayers(selected.clone()).into());
 						}
+					}
+					// If the user clicks on a layer that is in their current selection, go into the dragging mode.
+					// Otherwise enter the box select mode
+					if selected.iter().any(|path| intersection.contains(path)) {
+						data.layers_dragging = selected;
+						Dragging
 					} else {
-						responses.push_back(Operation::MountWorkingFolder { path: vec![] }.into());
-						data.layers_dragging = Vec::new();
+						responses.push_back(DocumentMessage::DeselectAllLayers.into());
+						data.box_id = Some(vec![generate_hash(&*responses, input, document.document.hash())]);
+						responses.push_back(
+							Operation::AddBoundingBox {
+								path: data.box_id.clone().unwrap(),
+								transform: DAffine2::ZERO.to_cols_array(),
+								style: style::PathStyle::new(Some(Stroke::new(Color::from_rgb8(0x31, 0x94, 0xD6), 2.0)), Some(Fill::none())),
```

Comment:
Let's make the stroke width 1.0 and the hex code `#00A8FF` (`0x00`, `0xA8`, `0xFF`).